nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -229,39 +229,13 @@ in
 
     systemd.user.services.dbus.wantedBy = [ "default.target" ];
 
-    programs.dconf.profiles.gdm =
-    let
-      customDconf = pkgs.writeTextFile {
-        name = "gdm-dconf";
-        destination = "/dconf/gdm-custom";
-        text = ''
-          ${optionalString (!cfg.gdm.autoSuspend) ''
-            [org/gnome/settings-daemon/plugins/power]
-            sleep-inactive-ac-type='nothing'
-            sleep-inactive-battery-type='nothing'
-            sleep-inactive-ac-timeout=0
-            sleep-inactive-battery-timeout=0
-          ''}
-        '';
+    programs.dconf.profiles.gdm.databases = [ "${gdm}/share/gdm/greeter-dconf-defaults" ] ++ lib.lists.optional cfg.gdm.autoSuspend {
+      org.gnome.settings-daemon.plugins.power = {
+        sleep-inactive-ac-type = "nothing";
+        sleep-inactive-battery-type = "nothing";
+        sleep-inactive-ac-timeout = 0;
```

Comment:
I wonder if we should require an explicit integer type (eg. `lib.gvariant.mkInt`). For example, `org.gnome.desktop.session`’s `idle-delay` requires `uint32` and people regularly trip on it, passing it a Nix number in home-manager, which implicitly gets turned into an int.

Alternative would be making the dconf module internal and creating a separate GSettings module that would also perform validation against schema.